### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by mwittgen**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,8 +21,7 @@ requirements:
     - pkg-config
 
   host:
-    - grpc-cpp
-    - fmt
+    - grpc-cpp <=1.42.0
```
Comment (beckermr) [inline recipe/meta.yaml:24]: We shouldn't put explicit pins in host. if you need an older build, this should go in a local conda_build_config.yaml in the recipe.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -31,7 +30,7 @@ requirements:
     - libprotobuf
 
 test:
-  commands: test -f ${PREFIX}/bin/bear
+  commands: test -f ${PREFIX}/bin/bear && bear -- test -d .
```
Comment (beckermr) [inline recipe/meta.yaml:33]:
```suggestion
  commands: 
   - test -f ${PREFIX}/bin/bear
   - bear -- test -d .
```

File: recipe/meta.yaml (hunk 1 above)
Comment (beckermr) [reply recipe/meta.yaml:24]: Also, if this results in a downgrade of the grpc-cpp pinning, then we need to merge to a branch.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -10,19 +10,19 @@ source:
...
+    - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - cmake
     - make
     - pkg-config
 
   host:
-    - grpc-cpp
-    - fmt
+    - grpc-cpp  {{ grpc_cpp }}
```
Comment (beckermr) [inline recipe/meta.yaml:25]: You don't need the jinja2 here I think.

Comment (mwittgen) [reply recipe/meta.yaml:25]: grpc-cpp is not a valid key. because of the `-`

**Comment by mwittgen** [comment]
The arm-osx queues fail, no clue what is going in. Looks like some cross compilation issue? Doesn't get past the cmake compiler detection.

**Comment by beckermr** [comment]
@erykoff is very good at arm builds.

**Comment by erykoff** [comment]
I think you might need:
```
  -DCMAKE_INSTALL_PREFIX=${PREFIX} \
  -DCMAKE_PREFIX_PATH=${PREFIX}    \
```
in your `build.sh`.

**Comment by mwittgen** [comment]
@erykoff the old version required MACOS SDK 10.12. After removing this requirement this builds successfully. 
10.12 was originally required as some library features were not available in older SDK versions.

**Comment by erykoff** [comment]
Well, now they're both broken, sigh.  If you tested locally with a native build I think you need the cmake args to make the cross-build work.  And I worry setting the sdk explicitly like that is going to break x86.

**Comment by erykoff** [comment]
oh my I made it even worse ...

**Comment by mwittgen** [comment]
No worries, somehow SDK related, I'll figure it out.

**Comment by mwittgen** [comment]
There are two issues: 
* the SDK pin to 10.12 is only needed for `osx Intel`.
* still unresolved, the  `grpc` code generation fails for `osx arm`, likely due to
cross compilation. Local building on native `osx arm` works.
The error calling `grpc` in the generation step in email is
`bad CPU type in executable`

**Comment by mwittgen** [comment]
Fixed the grpc osx arm cross compile.

**Pull request merged**